core/src/main/scala/epollcat/internal/ch/SocketHelpers.scala

```diff
@@ -120,6 +136,56 @@ private[ch] object SocketHelpers {
     new InetSocketAddress(inetAddr, port)
   }
 
+  def toInet6SocketAddress(
+      addr: Ptr[posix.netinet.in.sockaddr_in6]
+  ): InetSocketAddress = {
+    val port = posix.arpa.inet.htons(addr.sin6_port).toInt
+    val addrBytes = addr.sin6_addr.at1.asInstanceOf[Ptr[Byte]]
+    val inetAddr = InetAddress.getByAddress {
```

Comment:
This appears to work, which is a virtue.

I can explore if there is a more direct way of handling the ByteArray.  I think someone mentioned something along
this line a week or 10 days ago on Discourse.